test/integration/targets/apt/tasks/url-with-deps.yml

```diff
@@ -0,0 +1,64 @@
+- name: Install https transport for apt
+  apt:
+    name: apt-transport-https
+    state: latest
+    force: yes
+
+- name: Add RabbitMQ release signing key
+  apt_key:
+    url: https://ansible-ci-files.s3.amazonaws.com/test/integration/targets/setup_rabbitmq/rabbitmq-release-signing-key.asc
+    state: present
+
+- name: Add RabbitMQ Erlang repository
+  apt_repository:
+    repo: "deb https://dl.bintray.com/rabbitmq-erlang/debian {{ ansible_distribution_release }} erlang-20.x"
+    filename: 'rabbitmq-erlang'
+    state: present
+    update_cache: yes
+
+- name: Install deb file (rabbitmq-server) with dependencies from URL (check_mode)
+  apt:
+    deb: https://s3.amazonaws.com/ansible-ci-files/test/integration/targets/setup_rabbitmq/rabbitmq-server_3.7.14-1_all.deb
```

Comment:
Can we use a smaller package for this test? Ideally we'd use a custom created package with minimal deps required for the test. That would also help guarantee that the package wasn't already installed. We should still make sure to remove it at the start of the test just in case.

If not, we should at least make copies of the necessary files on S3 in a directory matching the test. Otherwise future changes to files for the `setup_rabbitmq` target may affect this test.